Plugins/NativeEngine/Source/NativeEngine.h

```diff
@@ -451,29 +323,28 @@ namespace Babylon
         void SetTypeArrayN(const UniformInfo& uniformInfo, const uint32_t elementLength, const arrayType& array);
 
         template<int size>
-        void SetIntArrayN(CommandBufferDecoder& decoder);
+        void SetIntArrayN(NativeDataStream::Reader& data);
 
         template<int size>
-        void SetFloatArrayN(CommandBufferDecoder& decoder);
+        void SetFloatArrayN(NativeDataStream::Reader& data);
 
         template<int size>
-        void SetFloatN(CommandBufferDecoder& decoder);
+        void SetFloatN(NativeDataStream::Reader& data);
 
         template<int size>
-        void SetMatrixN(CommandBufferDecoder& decoder);
+        void SetMatrixN(NativeDataStream::Reader& data);
 
         // Scratch vector used for data alignment.
         std::vector<float> m_scratch{};
 
         std::vector<Napi::FunctionReference> m_requestAnimationFrameCallbacks{};
 
-        Napi::Reference<Napi::ArrayBuffer> m_commandBuffer{};
-        Napi::Reference<Napi::ArrayBuffer> m_commandValidationBuffer{};
-        inline static ResourceTable<void(NativeEngine::*)(CommandBufferDecoder&)> s_commandTable{};
-
         const VertexArray* m_boundVertexArray{};
         FrameBuffer m_defaultFrameBuffer;
         FrameBuffer* m_boundFrameBuffer{};
         PerFrameValue<bool> m_boundFrameBufferNeedsRebinding;
+
+        // TODO: This should be changed to a non-owning ref once multi-update is available.
```

Comment:
What is multi-update? Should this be linked to either a GitHub issue or discussion with more info?

Reply:
There's already a pretty comprehensive discussion of this in one of our group chats, if I remember correctly.